src/main/java/core/basesyntax/dao/impl/CommentDaoImpl.java

```diff
@@ -12,21 +15,60 @@ public CommentDaoImpl(SessionFactory sessionFactory) {
 
     @Override
     public Comment create(Comment entity) {
-        return null;
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = factory.openSession();
+            transaction = session.beginTransaction();
+            session.save(entity);
+            transaction.commit();
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+                throw new RuntimeException("Can't save comment " + entity + " to DB!!! ", e);
+            }
```

Comment:
```suggestion
            if (transaction != null) {
                transaction.rollback();
            }
             throw new RuntimeException("Can't save comment " + entity + " to DB!!! ", e);
```

what if transaction == null? (Fix in all places)

Reply:
+